SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -595,6 +597,17 @@ Buffer {
 		^[\b_query, bufnum]
 	}
 
+	setSampleRate { arg value;
+		// sends the message first to check sanity of sr before changing the langage-side instance variable
+		server.listSendMsg(this.setSampleRateMsg(value));
+		sampleRate = value !? _.asFloat;
```

Comment:
If one calls `buf.setSampleRate(0)` or `buf.setSampleRate` this will set client-side sampleRate to 0 or nil. Then, when asking for `buf.duration` one would get either `inf` or `ERROR: binary operator '/' failed` respectively.

Perhaps instead of setting sampleRate, we should call this.updateInfo(action) and add action to args?

Reply:
for this one, should we do that replacement there and then also in setSampleRateMsg, or just in the latter, or just in the former?

Reply:
It's true that new, alloc and allocRead used to cache info before getting a reply from the server. It's also true that all the `read` methods send a "/query" msg at the end to update sclang. I don't know what would be best: to send an additional message (as I suggested), or to cache directly the value we reasonably assume the server would set.
In any case, for the scope of this PR, I think we should cache the assumed value in *new and *alloc (like they were doing before with `.sampleRate_(server.sampleRate)`, only this time it would be `if(value.isNil or: {value <= 0}) { server.sampleRate } { sampleRate }`)

Then for .setSampleRate we could stick to the same pattern.

Reply:
ok I think I did what you wanted, and to make it clear, the last commit is that only dif:

1. added check and cache in *new
2. added check and cache in *alloc
3. added check and cache in *allocconsecutive
4. removed the check and casting in allocMsg
5. added the check and cache in setSampleRate
6. did it too in setSampleRateMsg in case it is called directly

let me know what I did wrong.